File: Editorial/2078_Two_Furthest_Houses_With_Different_Colors/JavaScript/2_Greedy.js
var maxDistance = function(colors) {
    const n = colors.length;
    let ans = 0;
    for (let i = 0; i < n - 1; i++) {
        if (colors[i] !== colors[n - 1]) {
            ans = Math.max(ans, i, n - 1 - i);
        }
    }
    return ans;
};

File: Editorial/2001-2100/2078_Two_Furthest_Houses_With_Different_Colors/CSharp/2_Greedy.cs
public class Solution {
    public int MaxDistance(int[] colors) {
        int n = colors.Length;
        int ans = 0;
        for (int i = 0; i < n - 1; i++) {
            if (colors[i] != colors[n - 1]) {
                ans = Math.Max(ans, Math.Max(i, n - 1 - i));
            }
        }
        return ans;
    }
}
